source/DurationPicker.js

```diff
@@ -0,0 +1,733 @@
+	* display is configurable using template.
+	*
+	* @class moon.DurationPicker
+	* @ui
+	* @public
+	*/
+
+	enyo.kind(
+		/** @lends moon.DurationPicker.prototype */ {
+
+		/**
+		* @private
+		*/
+		name: 'moon.DurationPicker',
+
+		/**
+		* @private
+		*/
+		kind: 'moon.ExpandableListItem',
+
+		/**
+		* @private
+		*/
+		defaultKind: 'enyo.Control',
+
+		/**
+		* @private
+		*/
+		classes: 'moon-expandable-picker',
+
+		/**
+		* @private
+		*/
+		handlers: {
+			//* Handler for _onChange_ events coming from constituent controls
+			onChange: 'handleChangeEvent'
+		},
+
+		events: {
+			/**
+			* {@link moon.DurationPicker#event:onDurationChange}
+			*/
+			onDurationChange: '',
+
+			/**
+			* {@link moon.DurationPicker#event:onCountdownExpired}
+			*/
+			onPickerExpired: ''
+		},
+
+		/**
+		* @public
+		* @lends moon.DurationPicker.prototype
+		*/
+		published: {
+
+			/**
+			* Text to be displayed as the current value if no item is currently selected
+			*
+			* @type {String}
+			* @default ''
+			* @public
+			*/
+			noneText: '',
+
+			/**
+			* The value of the picker, expressed as a standard JavaScript 
+			* object.
+			*
+			* @type {String}
+			* @default null
+			* @public
+			*/
+			value: null,
+
+			/**
+			* Optional label for hour
+			*
+			* @type {String}
+			* @default moon.$L('hour')
+			* @public
+			*/
+			hourText: moon.$L('hour'),		// i18n 'HOUR' label in moon.DurationPicker widget
+
+			/**
+			* Optional label for minute
+			*
+			* @type {String}
+			* @default moon.$L('minute')
+			* @public
+			*/
+			minuteText: moon.$L('minute'),		// i18n 'MINUTE' label in moon.DurationPicker widget
+
+			/**
+			* Optional label for second
+			*
+			* @type {String}
+			* @default moon.$L('second')
+			* @public
+			*/
+			secondText: moon.$L('second'),		// i18n 'SECOND' label in moon.DurationPicker widget
+
+			/**
+			* The template format of the picker
+			*
+			* @type {String}
+			* @default ''
+			* @public
+			*/
+			template: '',
+
+			/**
+			* Enable and disable the timer feature of duration picker
+			*
+			* @type {Boolean}
+			* @default true
+			* @public
+			*/
+			enableTimer: true
+		},
+
+		/**
+		* @type {Boolean}
+		* @default false
+		* @private
+		*/
+		countdown: false,
+
+		/**
+		* @type {String}
+		* @default null
+		* @private
+		*/
+		timer: null,
+
+		/**
+		* @type {Object}
+		* @default {}
+		* @private
+		*/
+		valueArray: {},
+
+		/**
+		* @type {Array}
+		* @default []
+		* @private
+		*/
+		templateArray: [],
+
+		/**
+		* Optional label for Start Button
+		*
+		* @type {String}
+		* @default moon.$L('Start')
+		* @private
+		*/
+		startConent: moon.$L('Start'),
+
+		/**
+		* Optional label for Pause Button
+		*
+		* @type {String}
+		* @default moon.$L('Pause')
+		* @private
+		*/
+		pauseConent: moon.$L('Pause'),
+
+		/**
+		* @type {Boolean}
+		* @default true
+		* @private
+		*/
+		pauseDisabled: true,
+
+		/**
+		* @private
+		*/
+		observers: {
+			'changeCountdown': ['countdown']
```

Comment:
Do we need to explicitly specify this observer, or could we instead just implement a `countdownChanged` method?